### ISSUE TITLE
Incorrect GRE scans in sub-NYU1

### ISSUE THREAD

**Issue by Kyota-exe** [spinal-cord-7t/coil-qc-code#37]
According to @evaalonsoortiz, the GRE scans in sub-NYU1 are from another project. The GRE scans in sub-NYU1 on the OpenNeuro database must thus be replaced with the correct scans from this project.

**Comment by Kyota-exe** [comment]
Upon closer inspection, it seems all MtSinai subjects except for sub-NYU4 have the wrong GRE scans.

I found the DICOMs for the correct GRE scans, but I am encountering an issue performing the conversion to NIFTI. Running `dcm2niix` with the same arguments that I have used with success for GRE scans for other subjects does not properly separate the DICOMs into multiple NIFTI files (one for each channel). Instead, it outputs a single NIFTI file that contains all the channels in the fourth dimension.

The following is the output from `dcm2niix` when I attempt the conversion of these scans:
<pre>
> dcm2niix -z y -f %s_%p -m o -o "./out" .\gre_2D_2mmIso_RefV450_uncombined_ND\
Chris Rorden's dcm2niiX version v1.0.20240202  (JP2:OpenJPEG) (JP-LS:CharLS) MSC1938  (64-bit Windows)
Found 2001 DICOM file(s)
Convert 44 DICOM as ./out\29_gre_2D_2mmIso_RefV450_uncombined (144x192x44x1)
Warning: Images sorted by instance number [0020,0013](1..44), but AcquisitionTime [0008,0032] suggests a different order (134747..134747)
Warning: Unable to determine slice direction: please check whether slices are flipped
Convert 946 DICOM as ./out\30_gre_2D_2mmIso_RefV450_uncombined_ph (144x192x43x22)
Warning: Instance Number (0020,0013) order is not spatial.
Warning: Interslice distance varies in this volume (incompatible with NIfTI format).
Warning: Missing images? Expected 43 images, but instance number (0020,0013) ranges from 44 to 1
Convert 43 DICOM as ./out\31_gre_2D_2mmIso_RefV450_uncombined_ph (144x192x43x1)
Warning: Images sorted by instance number [0020,0013](1..44), but AcquisitionTime [0008,0032] suggests a different order (134747..134747)
Warning: Unable to determine slice direction: please check whether slices are flipped
<b>Convert 968 DICOM as ./out\28_gre_2D_2mmIso_RefV450_uncombined (144x192x44x22)</b>
Conversion required 7.609000 seconds.
</pre>

The 22 scans in the 4th dimension in **(144x192x44x22)** correspond to the 22 channels that we want to obtain as separate files.

**Comment by Kyota-exe** [comment]
I found the tool `fslsplit` which allowed me to split the scan into 22 different channel scans as desired, but it seems I am unable to open any of them in `fsleyes`, which indicates something may be wrong with them. I get the error `numpy.linalg.LinAlgError: Singular matrix`. The relevant data (the original scan and the split scans) can be found here: https://drive.google.com/file/d/1LHpLRVmdiWZCXBZFYsO5erYpgfxFoTg-/view?usp=sharing

I will continue to look into this.

**Comment by Kyota-exe** [comment]
The `Singular matrix` error seems to be the same error @evaalonsoortiz gets in issue #61 (see the screenshot of the error she gets when attempting to open the file in fsleyes): https://github.com/spinal-cord-7t/coil-qc-code/issues/61#issue-2273813610. The `dcm2niix` output for both conversions (the DICOMs from #61 and from this issue) includes the warning `Warning: Unable to determine slice direction: please check whether slices are flipped`. 

Maybe there is a way to flip the slice directions in the DICOMs before using `dcm2niix`? Does anyone know if this is something that has been done in the past to mitigate an issue like this?

**Comment by dpapp86** [comment]
@evaalonsoortiz @Kyota-exe 

Sorry for doing this at such an inopportune time. I am working from the data I have, so part of this might be not valid.

My "gre_2D_2mmIso_RefV450_uncombined_ND" folder has over 900 files. This is because it did not only contain the uncombined MAG files, but everything. 

So my solution to this was:
1, Open the folder itself in a DICOm viewer that can handle multiple series (Osirix, Horos, MicroDICOM, etc). For this specific folder, this was four series, of the following sizes:
44 slices, Magnitude (A)
44X22 slices, Magnitude (B)
44 slices, phase (C)
44X22 slices, Magnitude (D)

I deleted everything apart from B. I was left with 44x22, that is 968 files, corresponding to 44 slices of the GRE scan and 22 channels of the coil.

2, I converted it using dcm2niix (Most recent version, Ubunutu)

3, I got the "Singular matrix" error referenced above.
3b, I managed to open the nifti file in MRICrom and ITK SNAP (windows, because they use qform to visualise)

4, I opened the header using fslinfo and fslhd, and noticed that qform and sform were not the same. This is the ususal cause of "singular matrix" errors

5, I ran `fslorient -copyqform2sform B.nii`

6, FSLeyes could now open the file correctly.

**Comment by evaalonsoortiz** [comment]
Do you have the DICOMS for all 3 subjects? Could you do this for all of them? If so, then @Kyota-exe could upload the proper nifits to the data repo.

**Comment by dpapp86** [comment]
I only have data for myself, Laura and Spinoza. If you point me at Rob's data, I can download it.

With our without Distortion Correction?

**Comment by dpapp86** [comment]
Example without fixing headers here:
https://drive.google.com/drive/folders/13q_HCEvhgC3Op_WjDeiKyU25QUXozsib?usp=drive_link

Let me know if this is sufficient, and what edits you'd like if its not

**Comment by evaalonsoortiz** [comment]
The slices seem to have some intensity variation across them:

<img width="1160" alt="Screenshot 2024-12-19 at 2 07 22 PM" src="https://github.com/user-attachments/assets/3fb47f2e-392b-4095-9f53-6a2b83b7dca9" />

**Comment by dpapp86** [comment]
My best guess is that here, we see the issue where the slice and coil
dimensions are populated correctly, in the sense that there is no mixing
dimensions, there is still a mix in ordering according to channels.

So while slice 1 might show slice1,coil1, slice 10 might show slice10,coil
5.

Sadly, the anonymisation used by the university stripped a lot of useful
info.

Basically this is as good as it gets

**Comment by dpapp86** [comment]
I'll give a try with explicitly sorting by dicoms tags.

**Comment by dpapp86** [comment]
@evaalonsoortiz 

OK, I have looked into _all_ the tags of the data (including hidden/proprietary ones).

I have read the VB17 DICOM conformance statements.

The in-vivo data is simply missing required DICOM tags. There should be a tag to indicate what coil element the data has been acquired with, which is tag (0051,100F). The anonymization procedure has scrubbed this tag from the in-vivo data (I have access to myself and Laura), but not from Spinoza.

So the only tag that tells any converter anything is "Instancenumber" and the like, which shows that slice X has 22 instances associated with it, but it has no way to sort which of those 22 instances reflects which coil.

In summary, the linked file is as good as it gets.

**Comment by evaalonsoortiz** [comment]
Thanks for your efforts Daniel!
